addons/l10n_it_edi_doi/models/account_move.py

```diff
@@ -0,0 +1,238 @@
+        for move in self:
+            if move.state != 'draft' or not move.l10n_it_edi_doi_use_declaration_of_intent:
+                move.l10n_it_edi_doi_declaration_of_intent_id = False
+                continue
+
+            company = move.company_id
+            partner = move.commercial_partner_id
+            date = move.l10n_it_edi_doi_declaration_of_intent_date
+            currency = move.currency_id
+
+            # Avoid a query or changing a manually set declaration of intent
+            # (if the declaration is still valid).
+            current_declaration = move.l10n_it_edi_doi_declaration_of_intent_id
+            if current_declaration and not current_declaration._get_validity_warnings(company, partner, currency, date):
+                continue
+
+            declaration = self.env['l10n_it_edi_doi.declaration_of_intent']\
+                ._fetch_valid_declaration_of_intent(company, partner, currency, date)
+            move.l10n_it_edi_doi_declaration_of_intent_id = declaration
+
+    @api.depends('tax_totals', 'move_type')
+    def _compute_l10n_it_edi_doi_amount(self):
+        """
+        Consider all the lines in self that belong to declaration of intent `declaration`
+        and have the special declaration of intent tax applied.
+        This function computes the signed sum of the price_total of all those lines
+        (the tax amount of the lines is always 0).
+        The move_type determines the sign: 1 (-1) for inbound (outbound) types.
+        """
+        inbound_move_types = self.env['account.move'].get_inbound_types(include_receipts=True)
+        for company_id, moves in self.grouped('company_id').items():
+            tax = company_id.l10n_it_edi_doi_declaration_of_intent_tax
+            if not tax:
+                moves.l10n_it_edi_doi_amount = 0
+                continue
+            for move in moves:
+                declaration = move.l10n_it_edi_doi_declaration_of_intent_id
+                if not declaration:
+                    move.l10n_it_edi_doi_amount = 0
+                    continue
+                declaration_lines = move.invoice_line_ids.filtered(
+                    # The declaration tax cannot be used with other taxes on a single line
+                    # (checked in `_post`)
+                    lambda line: line.tax_ids.ids == tax.ids
+                )
+                sign = 1 if move.move_type in inbound_move_types else -1
+                move.l10n_it_edi_doi_amount = sign * sum(declaration_lines.mapped('price_total'))
+
+    @api.depends('l10n_it_edi_doi_declaration_of_intent_id', 'l10n_it_edi_doi_amount')
+    def _compute_l10n_it_edi_doi_warning(self):
+        for move in self:
+            move.l10n_it_edi_doi_warning = ''
+            declaration = move.l10n_it_edi_doi_declaration_of_intent_id
+
+            show_warning = declaration \
+                and move.is_sale_document(include_receipts=False) \
+                and move.state != 'cancel'
```

Comment:
```suggestion
            show_warning = (
                declaration
                and move.is_sale_document(include_receipts=False)
                and move.state == 'draft'
            )
```
Same as before for the line continuation :wink: 
Also, I think you only want to show the warning on draft invoices, it does not matter much once the invoice is posted IMO.

Reply:
I changed the line continuation.
I will re-check where we want to show the warnings

Reply:
I checked. We want the warning also on posted invoice. It informs the user sth. may be wrong.